File: smart-fridge-java-backend/src/main/java/css/ch/smartfridgejavabackend/auth/controller/AuthController.java
package css.ch.smartfridgejavabackend.auth.controller;

import css.ch.smartfridgejavabackend.auth.dto.AuthResponse;
import css.ch.smartfridgejavabackend.auth.dto.LoginRequest;
import css.ch.smartfridgejavabackend.auth.dto.SignupRequest;
import css.ch.smartfridgejavabackend.auth.service.AuthService;
import css.ch.smartfridgejavabackend.auth.service.JwtService;
import css.ch.smartfridgejavabackend.user.dto.UserResponseDTO;
import css.ch.smartfridgejavabackend.user.services.UserService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {
    private final AuthService authService;
    private final UserService userService;
    private final JwtService jwtService;

    @PostMapping("/signup")
    public ResponseEntity<AuthResponse> signup(@Valid @RequestBody SignupRequest signupRequest) {
        UserResponseDTO userDto = userService.createUser(signupRequest);
        String token = jwtService.generateToken(userDto.id());

        return ResponseEntity.status(HttpStatus.CREATED).body(new AuthResponse(token));
    }

    @PostMapping("/login")
    public ResponseEntity<AuthResponse> login(@Valid @RequestBody LoginRequest loginRequest) {
        String userId = authService.authenticate(loginRequest.email(), loginRequest.password());
        String token = jwtService.generateToken(userId);

        return ResponseEntity.ok(new AuthResponse(token));
    }
}

File: smart-fridge-backend/src/main/kotlin/ch/bbzw/smartfridgebackend/auth/controller/AuthController.kt
package ch.bbzw.smartfridgebackend.auth.controller

import ch.bbzw.smartfridgebackend.auth.dto.AuthResponse
import ch.bbzw.smartfridgebackend.auth.dto.LoginRequest
import ch.bbzw.smartfridgebackend.auth.dto.SignupRequest
import ch.bbzw.smartfridgebackend.auth.service.AuthService
import ch.bbzw.smartfridgebackend.auth.service.JwtService
import ch.bbzw.smartfridgebackend.user.service.UserService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class AuthController(
    private val authService: AuthService,
    private val userService: UserService,
    private val jwtService: JwtService
) {

    @PostMapping("/signup")
    fun signup(@Valid @RequestBody signupRequest: SignupRequest): ResponseEntity<AuthResponse> {
        val userDto = userService.createUser(signupRequest)
        val token = jwtService.generateToken(userDto)

        return ResponseEntity.status(HttpStatus.CREATED).body(AuthResponse(token))
    }

    @PostMapping("/login")
    fun login(@Valid @RequestBody loginRequest: LoginRequest): ResponseEntity<AuthResponse> {
        val userId = authService.authenticate(loginRequest.email, loginRequest.password)
        val userDTO = userService.getUserById(userId)
        val token = jwtService.generateToken(userDTO)

        return ResponseEntity.ok(AuthResponse(token))
    }
}